### PR TITLE
test: add tests for valid electron module names

### PR DESCRIPTION
**Pull request by RaisinTen**
#### Description of Change

https://github.com/electron/electron/pull/35915 landed without any tests, so this change adds some. This also documents why these variations exist. I admit, it isn't the best place for documenting such a thing because it's not very accessible but I don't know what the best place is either, so suggestions welcome!

Signed-off-by: Darshan Sen <raisinten@gmail.com>

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation is changed or added

#### Release Notes

Notes: none 

cc @MarshallOfSound

### CODE REVIEW

**Review by codebytere — APPROVED** [review]

File: lib/common/reset-search-paths.ts [hunk 1]
```diff
@@ -42,6 +42,8 @@ const makeElectronModule = (name: string) => {
   Module._cache[name] = electronModule;
 };
 
+// These module aliases are for TypeScript purposes. It is intentional they
+// work everywhere, they're superficial for TypeScript only.
```
[the remark this answers is not in the document]
Comment (RaisinTen) [reply lib/common/reset-search-paths.ts:46]: @codebytere this is basically a copy-paste of Samuel's comment regarding what `electron/main`, `electron/common` and `electron/renderer` are for - https://github.com/electron/electron/pull/35915#issuecomment-1269893433

[the remark this answers is not in the document]
Comment (RaisinTen) [reply lib/common/reset-search-paths.ts]: @codebytere I have rephrased the comment, does it look better now?

**Review by jkleinsc — APPROVED** [review]

File: lib/common/reset-search-paths.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (codebytere) [reply lib/common/reset-search-paths.ts]: looks great - thanks!

**Review by miniak — CHANGES_REQUESTED** [review]
I think we should do this fist https://github.com/electron/electron/pull/35959

**Review by miniak — APPROVED** [review]

**Pull request merged**